hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/details/ArtifactDetailsLayout.java

```diff
@@ -246,29 +259,28 @@ public Button generateCell(final Table source, final Object itemId, final Object
                 return deleteIcon;
             }
         });
+
     }
 
     private void confirmAndDeleteArtifact(final Long id, final String fileName) {
-
-        final ConfirmationDialog confirmDialog = new ConfirmationDialog(i18n.getMessage("caption.delete.artifact.confirmbox"),
+        final ConfirmationDialog confirmDialog = new ConfirmationDialog(
+                i18n.getMessage("caption.delete.artifact.confirmbox"),
                 i18n.getMessage("message.delete.artifact", new Object[] { fileName }), i18n.getMessage("button.ok"),
                 i18n.getMessage("button.cancel"), ok -> {
                     if (ok) {
                         artifactManagement.deleteArtifact(id);
                         uINotification.displaySuccess(i18n.getMessage("message.artifact.deleted", fileName));
                         if (artifactUploadState.getSelectedBaseSwModuleId().isPresent()) {
-                            populateArtifactDetails(artifactUploadState.getSelectedBaseSwModuleId().get(),
-                                    HawkbitCommonUtil.getFormattedNameVersion(
-                                            artifactUploadState.getSelectedBaseSoftwareModule().get().getName(),
-                                            artifactUploadState.getSelectedBaseSoftwareModule().get().getVersion()));
+                            final Optional<SoftwareModule> softwareModule = getSelectedSoftwareModule();
```

Comment:
The value of the selected software module might have changed at this point. You can't use the member variable here.